From fd2ab4db2cc38720ad50069532eba6a91191765c Mon Sep 17 00:00:00 2001
From: Michael S. Tsirkin <mst@redhat.com>
Date: Mon, 24 May 2010 09:28:42 -0300
Subject: [PATCH 1/7] virtio: invoke set_features on load

RH-Author: Michael S. Tsirkin <mst@redhat.com>
Message-id: <6568b694535079f61bfa34fa52bf5e92fbca229d.1274691899.git.mst@redhat.com>
Patchwork-id: 9481
O-Subject: [RHEL6.0 PATCH 2/2] virtio: invoke set_features on load
Bugzilla: 595263
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>

After migration, vhost was not getting features
acked because set_features callback was never invoked.
The fix is just to invoke that callback.

Reported-by: David L Stevens <dlstevens@us.ibm.com>
Signed-off-by: Michael S. Tsirkin <mst@redhat.com>
Tested-by: David L Stevens <dlstevens@us.ibm.com>
Cherry-picked from upstream fae054b070143a60ce40671470292efc2dc58a49
---
 hw/virtio.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 hw/virtio.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/hw/virtio.c b/hw/virtio.c
index 3e09acd..6de9aa3 100644
--- a/hw/virtio.c
+++ b/hw/virtio.c
@@ -692,6 +692,8 @@ int virtio_load(VirtIODevice *vdev, QEMUFile *f)
                 features, supported_features);
         return -1;
     }
+    if (vdev->set_features)
+        vdev->set_features(vdev, features);
     vdev->guest_features = features;
     vdev->config_len = qemu_get_be32(f);
     qemu_get_buffer(f, vdev->config, vdev->config_len);
-- 
1.7.0.3

